org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/ZigBeeDataStore.java

```diff
@@ -0,0 +1,167 @@
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.openhab.binding.zigbee.ZigBeeBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.xml.PrettyPrintWriter;
+import com.thoughtworks.xstream.io.xml.StaxDriver;
+import com.zsmartsystems.zigbee.IeeeAddress;
+import com.zsmartsystems.zigbee.database.ZclAttributeDao;
+import com.zsmartsystems.zigbee.database.ZclClusterDao;
+import com.zsmartsystems.zigbee.database.ZigBeeEndpointDao;
+import com.zsmartsystems.zigbee.database.ZigBeeNetworkDataStore;
+import com.zsmartsystems.zigbee.database.ZigBeeNodeDao;
+import com.zsmartsystems.zigbee.zdo.field.BindingTable;
+import com.zsmartsystems.zigbee.zdo.field.NodeDescriptor.FrequencyBandType;
+import com.zsmartsystems.zigbee.zdo.field.NodeDescriptor.MacCapabilitiesType;
+import com.zsmartsystems.zigbee.zdo.field.NodeDescriptor.ServerCapabilitiesType;
+import com.zsmartsystems.zigbee.zdo.field.PowerDescriptor.PowerSourceType;
+
+/**
+ * Serializes and deserializes the ZigBee network state.
+ *
+ * @author Chris Jackson
+ */
+public class ZigBeeDataStore implements ZigBeeNetworkDataStore {
+    /**
+     * The logger.
+     */
+    private final static Logger logger = LoggerFactory.getLogger(ZigBeeDataStore.class);
+
+    private final String networkStateFilePath;
+
+    public ZigBeeDataStore(String networkId) {
+        networkStateFilePath = ConfigConstants.getUserDataFolder() + "/" + ZigBeeBindingConstants.BINDING_ID + "/"
+                + networkId;
+
+        final File folder = new File(networkStateFilePath);
+        // Create the path for serialization.
+        if (!folder.exists()) {
+            logger.debug("Creating ZigBee persistence folder {}", networkStateFilePath);
+            if (!folder.mkdirs()) {
+                logger.error("Error while creating ZigBee persistence folder {}", networkStateFilePath);
+            }
+        }
+    }
+
+    private XStream openStream() {
+        XStream stream = new XStream(new StaxDriver());
+        stream.alias("ZigBeeNode", ZigBeeNodeDao.class);
+        stream.alias("ZigBeeEndpoint", ZigBeeEndpointDao.class);
+        stream.alias("ZclCluster", ZclClusterDao.class);
+        stream.alias("ZclAttribute", ZclAttributeDao.class);
+        stream.alias("MacCapabilitiesType", MacCapabilitiesType.class);
+        stream.alias("ServerCapabilitiesType", ServerCapabilitiesType.class);
+        stream.alias("PowerSourceType", PowerSourceType.class);
+        stream.alias("FrequencyBandType", FrequencyBandType.class);
+        stream.alias("BindingTable", BindingTable.class);
+        stream.alias("IeeeAddress", BindingTable.class);
+        stream.registerConverter(new IeeeAddressConverter());
+        return stream;
+    }
+
+    private File getFile(IeeeAddress address) {
+        return new File(networkStateFilePath + address + ".xml");
```

Comment:
There is a `File.separator` missing between the `networkStateFilePath` and the `address`